l1-contracts/deploy-scripts/DeployL1CoreContracts.s.sol

```diff
@@ -0,0 +1,684 @@
+
+        (
+            addresses.bridgehub.chainAssetHandlerImplementation,
+            addresses.bridgehub.chainAssetHandlerProxy
+        ) = deployTuppWithContract("L1ChainAssetHandler", false);
+        setBridgehubParams();
+
+        updateOwners();
+
+        saveOutput(outputPath);
+    }
+
+    function setBridgehubParams() internal {
+        IBridgehub bridgehub = IBridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.startBroadcast(msg.sender);
+        bridgehub.addTokenAssetId(bridgehub.baseTokenAssetId(config.eraChainId));
+        bridgehub.setAddresses(
+            addresses.bridges.l1AssetRouterProxy,
+            ICTMDeploymentTracker(addresses.bridgehub.ctmDeploymentTrackerProxy),
+            IMessageRoot(addresses.bridgehub.messageRootProxy),
+            addresses.bridgehub.chainAssetHandlerProxy
+        );
+        vm.stopBroadcast();
+        console.log("SharedBridge registered");
+    }
+
+    function updateSharedBridge() internal {
+        IL1AssetRouter sharedBridge = IL1AssetRouter(addresses.bridges.l1AssetRouterProxy);
+        vm.broadcast(msg.sender);
+        sharedBridge.setL1Erc20Bridge(IL1ERC20Bridge(addresses.bridges.erc20BridgeProxy));
+        console.log("SharedBridge updated with ERC20Bridge address");
+    }
+
+    function setL1NativeTokenVaultParams() internal {
+        IL1AssetRouter sharedBridge = IL1AssetRouter(addresses.bridges.l1AssetRouterProxy);
+        IL1Nullifier l1Nullifier = IL1Nullifier(addresses.bridges.l1NullifierProxy);
+        // Ownable ownable = Ownable(addresses.bridges.l1AssetRouterProxy);
+        vm.broadcast(msg.sender);
+        sharedBridge.setNativeTokenVault(INativeTokenVault(addresses.vaults.l1NativeTokenVaultProxy));
+        vm.broadcast(msg.sender);
+        l1Nullifier.setL1NativeTokenVault(IL1NativeTokenVault(addresses.vaults.l1NativeTokenVaultProxy));
+        vm.broadcast(msg.sender);
+        l1Nullifier.setL1AssetRouter(addresses.bridges.l1AssetRouterProxy);
+
+        vm.broadcast(msg.sender);
+        IL1NativeTokenVault(addresses.vaults.l1NativeTokenVaultProxy).registerEthToken();
+    }
+
+    function updateOwners() internal {
+        vm.startBroadcast(msg.sender);
+
+        IBridgehub bridgehub = IBridgehub(addresses.bridgehub.bridgehubProxy);
+        IOwnable(address(bridgehub)).transferOwnership(addresses.governance);
+        bridgehub.setPendingAdmin(addresses.chainAdmin);
+
+        IL1AssetRouter sharedBridge = IL1AssetRouter(addresses.bridges.l1AssetRouterProxy);
+        IOwnable(address(sharedBridge)).transferOwnership(addresses.governance);
+
+        ICTMDeploymentTracker ctmDeploymentTracker = ICTMDeploymentTracker(
+            addresses.bridgehub.ctmDeploymentTrackerProxy
+        );
+        IOwnable(address(ctmDeploymentTracker)).transferOwnership(addresses.governance);
+
+        vm.stopBroadcast();
+        console.log("Owners updated");
+    }
+
+    function saveOutput(string memory outputPath) internal virtual {
```

Comment:
just to double check, do we really update all this data? 
the amount of deployments looks substantially lower